cmd-tool/statefulset.lua

```diff
@@ -0,0 +1,77 @@
+local StatefulSet = {}
+
+function StatefulSet.captureCommandOutput(namespace,workloadName)
+    local command = "kubectl get statefulset.apps.kruise.io -n " .. namespace .. " -o yaml"
+
+    if workloadName ~= nil then
+        command = command .. " " .. workloadName
+    end
+    
+    local handle = io.popen(command)
+    local output = handle:read("*a")
+    local success, exit_reason, exit_code = handle:close()
+
+    if not success then
+        return nil, exit_reason, exit_code
+    end
+
+    return output
+end
+
+function StatefulSet.checkHealthWithTimeout(namespace,workloadName,timeout)
+    local lyaml = require("lyaml")
+
+    local function checkStatus()
+        local output = StatefulSet.captureCommandOutput(namespace,workloadName)
+        local obj, err = lyaml.load(output)
+
+        local hs = { status = "Progressing", message = "Waiting for initialization" }
+
+        if obj then
+            
+            local items = obj.items or { obj }
+
+            for _, item in ipairs(items) do
+
+                if item.metadata.generation == item.status.observedGeneration then
+                    if item.spec.updateStrategy.rollingUpdate.paused == true or not item.status.updatedAvailableReplicas then
+                        hs.status = "Suspended"
+                        hs.message = "Statefulset is paused"
+                    elseif item.spec.updateStrategy.rollingUpdate.partition ~= 0 and item.metadata.generation > 1 then
+                        if item.status.updatedAvailableReplicas == (item.status.replicas - item.spec.updateStrategy.rollingUpdate.partition) then
+                            hs.status = "Healthy"
```

Comment:
i think this should not be healthy, it should be suspended. And if the condition does not match, the status should be degraded

Reply:
i think the condition should return the status as "Healthy" because it indicates that the rolling update has successfully updated the specified number of replicas.